src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/ReadWriteWalker.vb

```diff
@@ -135,14 +151,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Select Case variable.Kind
                 Case SymbolKind.Local
                     Dim local = DirectCast(variable, LocalSymbol)
-                    If Not (local.IsConst Or Me._currentMethodOrLambda = local.ContainingSymbol) Then
+                    If Not local.IsConst AndAlso Me._currentMethodOrLambda <> local.ContainingSymbol Then
                         Me._captured.Add(local)
                     End If
                 Case SymbolKind.Parameter
                     Dim param = DirectCast(variable, ParameterSymbol)
-                    If Not (Me._currentMethodOrLambda = param.ContainingSymbol) Then
+                    If Me._currentMethodOrLambda <> param.ContainingSymbol Then
                         Me._captured.Add(param)
                     End If
+                Case SymbolKind.RangeVariable
+                    Dim range = DirectCast(variable, RangeVariableSymbol)
+                    If Me._currentMethodOrLambda <> range.ContainingSymbol AndAlso
+                        (Me._currentQueryLambda Is Nothing OrElse ' might be Nothing in error scenarios
```

Comment:
Not at present; I added the test to prevent a crash in some other tests that operate in error scenarios. The only requirement here is not to crash, and those other tests cover that.